### PR TITLE
Multiple Ionizers per Species

### PR DESCRIPTION
**Pull request by ax3l**
Enables to set multiple ionizers per species. Closes and implements #925.

Changes the `alias` (particle flag) `ionizer` to `ionizers` to reflect a sequence needs to be given in the form of:

```C++
ionizers<
    MakeSeq_t<
        particles::ionization::BSIEffectiveZ< PIC_Electrons >,
        particles::ionization::ADKCircPol< PIC_Electrons >,
        particles::ionization::ThomasFermi< PIC_Electrons >
    >
>
```

I find this feature as important that I would still apply it to the current release branch.

### To Do

- [x] first RT test (LWFA, case 10)
- [x] depends on ~~#1998~~
- some nice comparisons on a foil
   - [x] setup: #2008 
   - [ ] bug in Thomas-Fermi: #2007

### CODE REVIEW

File: src/picongpu/include/particles/traits/GetIonizerList.hpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+
+namespace picongpu
+{
+namespace traits
+{
+    /** Returns a sequence with ionizers for a species
+     *
+     * Several ionization methods can be assigned to a species which are called
+     * in-order.
```
Comment (n01r) [inline src/picongpu/include/particles/traits/GetIonizerList.hpp]: better: *consecutively*

File: src/picongpu/include/particles/traits/GetIonizerList.hpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        // this now resolves the alias into the actual object type, a list of ionizers
+        using FoundIonizerList = typename PMacc::traits::Resolve< FoundIonizersAlias >::type;
+
+        using type = typename PMacc::OperateOnSeq<
+            FoundIonizerList,
+            bmpl::apply1<
+                bmpl::_1,
+                SpeciesType
+            >,
+            PMacc::compileTime::accessors::Type<>
```
Comment (n01r) [inline src/picongpu/include/particles/traits/GetIonizerList.hpp]: nice :smile:

File: src/picongpu/include/particles/traits/GetIonizerList.hpp (hunk 1 above)
Comment (ax3l) [inline src/picongpu/include/particles/traits/GetIonizerList.hpp]: btw, I hope `ForEach` does keep the order of the sequence it gets, right @psychocoderHPC ? ;)

**Comment by ax3l** [comment]
This PR is so far ready to be merged.

It is RT tested and I will post comparisons of sole to combined models in #2008 as soon as #2007 is fixed.

**Pull request merged**